### PR TITLE
Us 1146 2 fix voltage level creation in unbuilt substation

### PR DESCRIPTION
**Pull request by laurgarn**
Fix regressions occasioned by first part

### CODE REVIEW

**Review by archendel — CHANGES_REQUESTED** [review]
Test KO :
When creating a voltage level, if there are some results returned by the elasticsearch request and the user presses "enter" or clicks somewhere else on the screen, the input is replaced by the first option suggested by the autocomplete system

**Comment by laurgarn** [comment]
> Test KO : When creating a voltage level, if there are some results returned by the elasticsearch request and the user presses "enter" or clicks somewhere else on the screen, the input is replaced by the first option suggested by the autocomplete system

I checked out the commit before the merging of #713.
For in substation choice, as soon as the popup is open, the item nearest to the outer click will be selected.
For connection creation, the difference I notice is you can not longer input a "free" string or it will be emptied when leaving -> is it the misbehaviour you signal ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Tests: 
when there are no equipements of a particuliar type in the delete equipment dialog, the 'Loading ...' message remains displayed ...

File: src/components/dialogs/voltage-level-creation-dialog.js [hunk 1]
```diff
@@ -400,7 +403,7 @@ const VoltageLevelCreationDialog = ({
                 voltageLevelId,
                 voltageLevelName: voltageLevelName ? voltageLevelName : null,
                 nominalVoltage,
-                substationId: substation.id,
+                substationId: substation.id ?? substation,
```
[the remark this answers is not in the document]
Comment (laurgarn) [reply src/components/dialogs/voltage-level-creation-dialog.js:404]: Nice idea. Changed.

File: src/components/dialogs/connectivity-edition.js [hunk 2]
```diff
@@ -255,6 +257,7 @@ const ConnectivityEdition = ({
                     <Autocomplete
                         size="small"
                         freeSolo
+                        autoSelect
```
[the remark this answers is not in the document]
Comment (laurgarn) [reply src/components/dialogs/connectivity-edition.js]: Outdated code. Was an intermediate in transition to useAutocompleteField

File: src/utils/functions.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/**
```
[the remark this answers is not in the document]
Comment (laurgarn) [reply src/utils/functions.js:1]: Re-routed to another PR.

**Review by FranckLecuyer — APPROVED** [review]

**Review by archendel — APPROVED** [review]

**Pull request merged**